shared/src/main/scala/scala/util/parsing/input/OffsetPosition.scala

```diff
@@ -34,10 +34,14 @@ case class OffsetPosition(source: CharSequence, offset: Int) extends Position {
 
   private def genIndex: Array[Int] = {
     val lineStarts = new ArrayBuffer[Int]
-    lineStarts += 0
-    for (i <- 0 until source.length)
-      if (source.charAt(i) == '\n') lineStarts += (i + 1)
-    lineStarts += source.length
+    lineStarts += 0 // first line
+    for (i <- 0 until source.length) {
+      if (i >= 1 && source.charAt(i - 1) == '\n') // \n or \r\n
```

Comment:
the majority of the time the last character won't be a `\r` or `\n` might be worth binding `charAt(i - 1)` to a variable to save the extra lookup (dependant on the performance requirements of this code)